src/ifs/wl_seat.rs

```diff
@@ -883,13 +891,24 @@ impl WlSeatGlobal {
         client: &Rc<Client>,
         version: Version,
     ) -> Result<(), WlSeatError> {
+        let mut capabilities = POINTER | KEYBOARD;
+        let handlers = &self.state.input_device_handlers;
+        for (_, d) in handlers.borrow().iter() {
+            let dev = &d.data.device;
+            if dev.has_capability(InputDeviceCapability::Touch) {
+                capabilities |= TOUCH;
+                break;
+            }
+        }
```

Comment:
The capabilities of a seat should only consider the input devices connected to a seat. These capabilities can also change at runtime when a device is attached or detached and then they will have to be sent to the client again.